### PR TITLE
refactor: extract `<ControlLabel/>` component

### PR DESCRIPTION
**Pull request by ogonkov**

### CODE REVIEW

File: src/components/CheckedLabel/CheckedLabel.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import {DOMProps, QAProps} from '../types';
+import {block} from '../utils/cn';
+
+import './CheckedLabel.scss';
+
+type Props = PropsWithChildren<{
+    labelClassName?: string;
+    title?: string;
+    disabled?: boolean;
+    size?: 'm' | 'l';
```
Comment (korvin89) [inline src/components/CheckedLabel/CheckedLabel.tsx:11]: Maybe we should add one type for this sizes collection?

File: src/components/CheckedLabel/CheckedLabel.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import React, {forwardRef, PropsWithChildren, ReactNode} from 'react';
+import {DOMProps, QAProps} from '../types';
+import {block} from '../utils/cn';
+
+import './CheckedLabel.scss';
+
+type Props = PropsWithChildren<{
```
Comment (korvin89) [inline src/components/CheckedLabel/CheckedLabel.tsx:7]: Maybe text should added by property label/text

Comment (ogonkov) [reply src/components/CheckedLabel/CheckedLabel.tsx:7]: I think passing through `children` is more robust

File: src/components/CheckedLabel/types.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import {PropsWithChildren, ReactNode} from 'react';
+import {DOMProps, QAProps} from '../types';
+
+export type Sizes = 'm' | 'l';
+
+export type Props = PropsWithChildren<{
```
Comment (amje) [inline src/components/CheckedLabel/types.ts:6]: Let's use a full name for Sizes and Props, and `Sizes` should be `Size`

File: src/components/CheckedLabel/types.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    labelClassName?: string;
```
Comment (amje) [inline src/components/CheckedLabel/types.ts:7]: simply `className`, there is no more elements to add class to

File: src/components/CheckedLabel/CheckedLabel.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import React, {forwardRef} from 'react';
+import {block} from '../utils/cn';
+import {Props} from './types';
+
+import './CheckedLabel.scss';
+
+const b = block('checked-label');
+
+export const CheckedLabel = forwardRef<HTMLLabelElement, Props>(
```
Comment (amje) [inline src/components/CheckedLabel/CheckedLabel.tsx:9]: I would name the component as `ControlLabel`, checked word is too specific

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/CheckedLabel/types.ts (hunk 4 above)
Comment (ogonkov) [reply src/components/CheckedLabel/types.ts:7]: There is also `className`, inherited from `DOMProps`

File: src/components/CheckedLabel/CheckedLabel.tsx (hunk 5 above)
Comment (ogonkov) [reply src/components/CheckedLabel/CheckedLabel.tsx:9]: Not sure that this component will fit regular TextInput

**Review by amje — APPROVED** [review]

**Review by trabemz — APPROVED** [review]

**Review by korvin89 — APPROVED** [review]

**Pull request merged**